### PR TITLE
Role Assignments / Role Definitions

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Sorry this is one large PR, but everything depended on everything else :(

- **New Data Source:** `azurerm_role_definition`
- **New Resource:** `azurerm_role_assignment`
- **New Resource:** `azurerm_role_definition`

### CODE REVIEW

File: azurerm/data_source_builtin_role_definition.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+			},
+		},
+	}
+}
+
+func dataSourceArmBuiltInRoleDefinitionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).roleDefinitionsClient
+	name := d.Get("name").(string)
+	roleDefinitionIds := map[string]string{
+		"Contributor":               "/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c",
```
Comment (mbfrahry) [inline azurerm/data_source_builtin_role_definition.go]: Are these values (`Contributor`, `Owner`, Reader`, `VirtualMachineContributor`) specific to our account or azure in general?

Comment (tombuildsstuff) [reply azurerm/data_source_builtin_role_definition.go]: They're Azure in General - this Data Source is mostly there so users don't have to refer to GUIDs tbh

Comment (mbfrahry) [reply azurerm/data_source_builtin_role_definition.go]: Perfect!

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Pull request merged**